packages/database/src/core/CompoundWrite.ts

```diff
@@ -30,145 +30,170 @@ import { each } from './util/util';
  * to reflect the write added.
  */
 export class CompoundWrite {
-  constructor(private writeTree_: ImmutableTree<Node>) {}
+  constructor(public writeTree_: ImmutableTree<Node>) {}
 
   static empty(): CompoundWrite {
     return new CompoundWrite(new ImmutableTree(null));
   }
+}
 
-  addWrite(path: Path, node: Node): CompoundWrite {
-    if (path.isEmpty()) {
-      return new CompoundWrite(new ImmutableTree(node));
+export function addWrite(
```

Comment:
Can you prefix all these method names with `compoundWrite`? This has some benefits:
- All imports sort in order
- It is still possible to translate Android and iOS APIs to Web APIs
- There are no clashes and we don't have to think about how to deal with clashes on a case-by-case basis.
- 
I initially did not like this method naming either, but we updated all of Firestore to match this pattern. This created a very consistent method-naming convention across the SDK. See https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/local_store_impl.ts#L216

Reply:
Done. It's starting to look a little like Java now!